Simplenote/src/androidTest/java/com/automattic/simplenote/smoke/pages/BasePage.java

```diff
@@ -0,0 +1,312 @@
+package com.automattic.simplenote.smoke.pages;
+
+import android.content.Context;
+import android.os.Build;
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.ViewParent;
+
+import androidx.test.espresso.Espresso;
+import androidx.test.espresso.PerformException;
+import androidx.test.espresso.UiController;
+import androidx.test.espresso.ViewAction;
+import androidx.test.espresso.ViewInteraction;
+import androidx.test.espresso.action.ViewActions;
+import androidx.test.espresso.matcher.BoundedMatcher;
+import androidx.test.espresso.matcher.ViewMatchers;
+import androidx.test.espresso.util.HumanReadables;
+import androidx.test.espresso.util.TreeIterables;
+
+import com.automattic.simplenote.smoke.support.SupplierIdler;
+import com.automattic.simplenote.widgets.RobotoMediumTextView;
+import com.google.android.material.textfield.TextInputLayout;
+
+import org.hamcrest.BaseMatcher;
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.hamcrest.TypeSafeMatcher;
+
+import java.util.concurrent.TimeoutException;
+import java.util.function.Supplier;
+
+import static androidx.test.espresso.Espresso.onView;
+import static androidx.test.espresso.action.ViewActions.click;
+import static androidx.test.espresso.action.ViewActions.replaceText;
+import static androidx.test.espresso.assertion.ViewAssertions.matches;
+import static androidx.test.espresso.matcher.ViewMatchers.isAssignableFrom;
+import static androidx.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static androidx.test.espresso.matcher.ViewMatchers.isRoot;
+import static androidx.test.espresso.matcher.ViewMatchers.withEffectiveVisibility;
+import static androidx.test.espresso.matcher.ViewMatchers.withId;
+import static androidx.test.espresso.matcher.ViewMatchers.withText;
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.not;
+import static org.hamcrest.core.StringContains.containsString;
+
+public class BasePage {
+    protected void clickButton(Integer resourceId) {
+        onView(allOf(withId(resourceId), isDisplayed())).perform(click());
+    }
+
+    protected void clickButton(Integer resourceId, String text) {
+        onView(allOf(withId(resourceId), isDisplayed(), withText(text))).perform(click());
+    }
+
+    protected void enterTextInCustomInput(Integer resourceId, String text) {
+        getViewById(resourceId)
+                .perform(click())
+                .perform(replaceTextInCustomInput(text))
+                .perform(ViewActions.closeSoftKeyboard());
+    }
+
+    protected void enterText(Integer resourceId, String text) {
+        getViewById(resourceId)
+                .perform(click())
+                .perform(replaceText(text))
+                .perform(ViewActions.closeSoftKeyboard());
+    }
+
+    protected void checkTextOnViews(Integer resourceId, String text) {
```

Comment:
This method doesn't appear to be used anywhere in the project – does it need to be in this PR?

Reply:
👍